tcms/rpc/tests/test_testcasestatus.py

```diff
@@ -0,0 +1,57 @@
+from tcms.rpc.tests.utils import APITestCase
+from tcms.testcases.models import TestCaseStatus
+from tcms.tests.factories import (
+    ProductFactory,
+    TestCaseFactory,
+    TestPlanFactory,
+    UserFactory,
+    VersionFactory,
+)
+
+
+class TestFilterCaseStatus(APITestCase):
+    """ Tests the XML-RPC testcasestatus.filter method """
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.tester = UserFactory(username="great tester")
+        self.product = ProductFactory(name="StarCraft")
+        self.version = VersionFactory(value="0.1", product=self.product)
+        self.plan = TestPlanFactory(
+            name="Test case_status.filter",
+            author=self.tester,
+            product=self.product,
+            product_version=self.version,
+        )
+        self.case_status_1 = TestCaseStatus.objects.all()[0]
+        self.case_status_2 = TestCaseStatus.objects.all()[1]
+
+        self.test_case_1 = TestCaseFactory(
+            author=self.tester,
+            reviewer=self.tester,
+            default_tester=None,
+            plan=[self.plan],
+            case_status=self.case_status_1,
+        )
+        self.test_case_2 = TestCaseFactory(
+            author=self.tester,
+            reviewer=self.tester,
+            default_tester=None,
+            plan=[self.plan],
+            case_status=self.case_status_2,
+        )
+
+    def test_filter_case_status(self):
+        cases_status = self.rpc_client.TestCaseStatus.filter(
+            {"name": self.test_case_1.case_status.name}
+        )
+        self.assertIsNotNone(cases_status)
+        self.assertEqual(len(cases_status), 1)
+
+    def test_filter_case_status_by_id(self):
+        cases_status = self.rpc_client.TestCaseStatus.filter(
+            {"id": self.test_case_2.case_status.id}
+        )
+        status = cases_status[0]
+        self.assertEqual(status["name"], self.test_case_2.case_status.name)
```

Comment:
These scenarios are a bit redundant. To validate that the API method works you can filter by any criteria (or no criteria at all) and assert the result is a non-empty list. 

There are 4 statuses already populated in the DB, see `testcases/migrations/0001_initial.py`